services/web/server/src/simcore_service_webserver/garbage_collector/_core_guests.py

```diff
@@ -0,0 +1,236 @@
+                delete_tasks.append(task)
+
+            except ProjectNotFoundError:
+                logging.warning(
+                    "Project with %s was not found, skipping removal",
+                    f"{project_uuid=}",
+                )
+
+        else:
+            # Try to change the project owner and remove access rights from the current owner
+            _logger.debug(
+                "Transferring ownership of project %s from user %s to %s.",
+                "This project cannot be removed since it is shared with other users"
+                f"{project_uuid=}",
+                f"{user_id}",
+                f"{new_project_owner_gid}",
+            )
+            await replace_current_owner(
+                app=app,
+                project_uuid=project_uuid,
+                user_primary_gid=user_primary_gid,
+                new_project_owner_gid=new_project_owner_gid,
+                project=project,
+            )
+
+    # NOTE: ensures all delete_task tasks complete or fails fast
+    # can raise ProjectDeleteError, CancellationError
+    await asyncio.gather(*delete_tasks)
+
+
+async def remove_guest_user_with_all_its_resources(
+    app: web.Application, user_id: int
+) -> None:
+    """Removes a GUEST user with all its associated projects and S3/MinIO files"""
+
+    try:
+        user_role: UserRole = await get_user_role(app, user_id)
+        if user_role > UserRole.GUEST:
+            # NOTE: This acts as a protection barrier to avoid removing resources to more
+            # priviledge users
+            return
+
+        _logger.debug(
+            "Deleting all projects of user with %s because it is a GUEST",
+            f"{user_id=}",
+        )
+        await _delete_all_projects_for_user(app=app, user_id=user_id)
+
+        _logger.debug(
+            "Deleting user %s because it is a GUEST",
+            f"{user_id=}",
+        )
+        await delete_user_without_projects(app, user_id)
+
+    except (
+        DatabaseError,
+        asyncpg.exceptions.PostgresError,
+        ProjectNotFoundError,
+        UserNotFoundError,
+        ProjectDeleteError,
+    ) as error:
+        _logger.warning(
+            "Failed to delete guest user %s and its resources: %s",
+            f"{user_id=}",
+            f"{error}",
+        )
+
+
+async def remove_users_manually_marked_as_guests(
+    registry: RedisResourceRegistry, app: web.Application
+) -> None:
+    """
+    Removes all the projects MANUALLY marked as GUEST users in the system (i.e. does not include
+    those accessing via the front-end).
+    If the user defined a TEMPLATE, this one also gets removed.
+    """
+    redis_locks_client: Redis = get_redis_lock_manager_client(app)
+
+    # collects all users with registed sessions
+    alive_keys, dead_keys = await registry.get_all_resource_keys()
+
+    skip_users = set()
+    for entry in chain(alive_keys, dead_keys):
+        skip_users.add(int(entry["user_id"]))
```

Comment:
Basically here we want to avoid any user that currently holds some information in redis (i.e. the user-sessions registry)

Otherwise, anything in the system running and refering to users that has nothing in the registry should be garbage-collected. I think that is the logic behind this